### ISSUE TITLE
[Bug]: Missing axes limits auto-scaling support for LineCollection

### ISSUE THREAD

**Issue by carlosgmartin** [matplotlib/matplotlib#29219]
### Bug summary

Matplotlib is missing auto-scale support for LineCollection.

Related issues:
- https://github.com/matplotlib/matplotlib/issues/23317/
- https://github.com/matplotlib/matplotlib/pull/28403

### Code for reproduction

```Python
from matplotlib import pyplot as plt
from matplotlib.collections import LineCollection

lc = LineCollection([[(x, x ** 2) for x in range(5)]])
ax = plt.gca()
ax.add_collection(lc)
# ax.autoscale()  # need to manually call this
plt.show()
```

### Actual outcome

![image](https://github.com/user-attachments/assets/a229d253-a69a-45b4-8a57-bfdd71fd86a1)

### Expected outcome

![image](https://github.com/user-attachments/assets/48566b7a-73b5-470d-ade1-e869ee288da2)

### Operating system

macOS 14.6.1

### Matplotlib Version

3.9.3

### Matplotlib Backend

macosx

### Python version

3.12.7

### Jupyter version

7.2.2

### Installation

pip

**Comment by li-ruihao** [comment]
Hello, would this be a first good issue? This is my first time contributing to Matplotlib, can I take a look at this? I know issues will not be assigned and PRs are considered on a first come first serve basis, but I need to work on an issue for a school project so I am wondering how much learning curve this requires.

**Comment by QuLogic** [comment]
There is already a PR open to fix this, linked above. I have not looked at it myself to see what state it is in.

**Comment by rcomer** [comment]
Looking through the `axes._axes` module, the methods which add a collection all call `_request_autoscale_view` after `add_collection`.  So I think the _autolim_ parameter works to update the data limits, but the axises still need to be told those limits have changed.  This would _appear_ to suggest that the current behaviour is intentional but if nothing else we should add a docstring to `add_collection` and make the expected behaviour is clear.

**Comment by carlosgmartin** [comment]
@rcomer "Intentional" or not, the current behavior is not user-friendly and should be improved.

**Comment by rcomer** [comment]
@carlosgmartin I agree that the current behaviour is unlikely to be what anyone wants.  Indeed, if you plot something else that has autoscaling after adding the collection then it autoscales both.  Whether something gets autoscaled should not depend on whether you add something after.

```python
from matplotlib import pyplot as plt
from matplotlib.collections import LineCollection

lc = LineCollection([[(x, x ** 2) for x in range(5)]])
ax = plt.gca()
ax.add_collection(lc)
ax.plot([1, 0])
plt.show()
```

![image](https://github.com/user-attachments/assets/a3d9baf6-fd13-496a-9758-75a528b61efd)

However, it does appear to have been deliberately written this way and I think we should try to understand _why_ it was written this way.  Understanding that can help us scope what kind of task it would be to change it.

**Comment by rcomer** [comment]
If we simply add a call to `_request_autoscale_view` within `add_collection`, we get two test failures:

1. [test_path_collection](https://github.com/matplotlib/matplotlib/blob/671177c08613136fd5004092b8b56449d419c12a/lib/matplotlib/tests/test_backend_ps.py#L347-L362) where the current image is this:   
    ![scatter-expected_eps](https://github.com/user-attachments/assets/855d447e-0ec5-4940-aebf-40127653eb84)
    but the new behaviour gives this, because the y-axis is now much taller
    ![scatter_eps](https://github.com/user-attachments/assets/afe49c36-bc49-4696-b020-14055781ff26)
    This test was added relatively recently by @oscargus so perhaps he has an opinion on what to do with it.

2. [test_arc_in_collection](https://github.com/matplotlib/matplotlib/blob/671177c08613136fd5004092b8b56449d419c12a/lib/matplotlib/tests/test_patches.py#L938-L945) tests consistency between `add_collection` and `add_patch`.  We can make this one pass by also calling `_request_autoscale_view` within `add_patch`, but then we get about a dozen additional test failures...

**Comment by timhoffm** [comment]
We have a lazy limit update mechanism, but I don’t know the details and which part of the code are responsible for which aspects of that. I believe we have to clarify this to find out, what the right solution is. It might also be that this is currently not handled consistently across functions.

**Comment by oscargus** [comment]
> 1. This test was added relatively recently by @oscargus so perhaps he has an opinion on what to do with it.

The purpose of the test was to get coverage, so the test image is just randomly generated. My impression is though that everything is shown in the test image so the compression is not caused by things that are not drawn. I guess seeing the axis here would be interesting to understand what happens. (As the data is randomly generated, I guess I just tried things until I got a reasonable looking image.)

Note that this test was for a previously untested part of the ps-backend, so it may very well be that there is a bug in the ps-backend and that one should run the same code with other backends so see if that is the case...

**Comment by oscargus** [comment]
Running the code in the qt-backend gives a differently looking image. The scatter looks quite similar, but the `PathCollection` is rendered differently, larger size and no black edges. 

![image](https://github.com/user-attachments/assets/813e809b-13dc-4e79-a8ff-36728196966c)

Saving it as PDF leads to a similar image as the test image, but no black edges.

![image](https://github.com/user-attachments/assets/88f4e1f8-a232-45a3-95be-2f97a8db7f8a)

(both are screen shots)

So something is quite strange here...

**Comment by rcomer** [comment]
I think the black edges are just because the test runs in "classic" mode.

**Comment by oscargus** [comment]
> I think the black edges are just because the test runs in "classic" mode.

Could be. Documentation may lead one to believe differently: 
> default: [rcParams["patch.edgecolor"]] (default: 'black')

The return value of `get_edgevalue` is `array([], shape=(0, 4), dtype=float64)` though.

**Comment by oscargus** [comment]
I read up a bit in the source and there is `rcParam['patch.force_edgecolor']` which if `True`, the statement in the documentation is correct, otherwise the edge color is set to `'none'`

I can sort out a correction to the documentation if nothing else.

**Comment by QuLogic** [comment]
Seems to be fixed by #29958.
